symplyphysics/core/experimental/coordinate_systems/new_coordinate_systems.py

```diff
@@ -0,0 +1,355 @@
+from __future__ import annotations
+
+from typing import Optional, Sequence
+from functools import partial
+from sympy import (Expr, sqrt, Matrix, true, sin, cos, tan, Q, simplify, atan2, Symbol as SymSymbol,
+    S, ImmutableMatrix, Basic)
+from sympy.logic.boolalg import Boolean
+from symplyphysics import Symbol, units, clone_as_function, Function
+
+from ..miscellaneous import cacheit
+
+
+class BaseCoordinateSystem(Basic):  # type: ignore[misc]
+    _base_scalars: tuple[Symbol, Symbol, Symbol]
+
+    _cartesian_derivative_matrix: ImmutableMatrix
+    """See `BaseCoordinateSystem.cartesian_derivative_matrix`."""
+
+    _lame_coefficients: tuple[Expr, Expr, Expr]
+    """See `BaseCoordinateSystem.lame_coefficients`."""
+
+    _base_vector_matrix: ImmutableMatrix
+    """See `BaseCoordinateSystem.base_vector_matrix`."""
+
+    _wrt: Symbol
+    _base_scalar_functions: tuple[Function, Function, Function]
+    _diff_base_vector_matrix: ImmutableMatrix
+    """See `BaseCoordinateSystem.diff_base_vector_matrix`."""
+
+    @property
+    def base_scalars(self) -> tuple[Symbol, Symbol, Symbol]:
+        return self._base_scalars
+
+    @property
+    def base_scalar_functions(self) -> tuple[Function, Function, Function]:
+        return self._base_scalar_functions
+
+    def _hashable_content(self) -> tuple[Basic, ...]:
+        return ()
+
+    def __repr__(self) -> str:
+        return type(self).__qualname__
+
+    @cacheit
+    def __new__(
+        cls,
+        base_scalars: Optional[tuple[Symbol, Symbol, Symbol]] = None,
+        base_scalar_functions: Optional[tuple[Function, Function, Function]] = None,
+    ) -> BaseCoordinateSystem:
+        obj = super().__new__(cls)
+
+        if not base_scalars:
+            base_scalars = obj.generate_base_scalars()
+
+        q1, q2, q3 = base_scalars
+        obj._base_scalars = q1, q2, q3
+
+        a, b, c = base_scalars
+        x, y, z = obj.cartesian_transform(base_scalars)
+
+        obj._cartesian_derivative_matrix = ImmutableMatrix([
+            [x.diff(a), y.diff(a), z.diff(a)],
+            [x.diff(b), y.diff(b), z.diff(b)],
+            [x.diff(c), y.diff(c), z.diff(c)],
+        ])
+
+        try:
+            obj._lame_coefficients = obj.generate_lame_coefficients()
+        except NotImplementedError:
+            assumption = obj.assumption(base_scalars)
+
+            lame_coefficients = []
+
+            for i_row in range(3):
+                row = obj._cartesian_derivative_matrix.row(i_row)
+                lame_coefficient = sqrt(sum(elem**2 for elem in row)).refine(assumption).simplify()
+                lame_coefficients.append(lame_coefficient)
+
+            h1, h2, h3 = lame_coefficients
+            obj._lame_coefficients = h1, h2, h3
```

Comment:
This code looks like an overkill. Inherited class can decide to use this kind of lame coefficients generation. But base class should not try to guess what inherited class will do.